src/Raven.Server/Documents/TransactionOperationsMerger.cs

```diff
@@ -70,7 +70,7 @@ public abstract class MergedTransactionCommand
         /// Enqueue the command to be eventually executed. If the command implements
         ///  IDisposable, the command will be disposed after it is run and a tx is committed.
         /// </summary>
-        public async Task Enqueue(MergedTransactionCommand cmd)
+        public async ValueTask<bool> Enqueue(MergedTransactionCommand cmd)
```

Comment:
Even in waiting cases the allocation is done on the stack, so yes. That single one was 2Gb allocations in 1 minute test